Improve handling of INTEGER timestamps

RobAtticus opened the pull request:
Previously the default chunk time in microseconds was too large
for an INTEGER field. If a user is using integer, they are using
seconds (since both milli and microsecond overflow), so now we
convert the default to the appropriate number in seconds.

Further, we check to make sure the chunk time interval is not too
large for INTEGER so as to avoid confusing problems later when the
user goes to insert.

Address issue #150

cevian commented on sql/ddl_api.sql:
```diff
@@ -81,6 +84,22 @@ BEGIN
+    WHERE attrelid = main_table AND attname = time_column_name;
+
+    chunk_time_interval_actual := chunk_time_interval;
+    IF time_type = 'INTEGER'::REGTYPE AND chunk_time_interval = _timescaledb_internal.interval_to_usec('1 month') THEN
```
I think we need to take a different tack. Change the default value for the `chunk_time_interval` parameter to be NULL and then set it to 1 month if it is NULL (either in usec or sec depending on column type). Right now this conversion will happen if the user passes in 1 month explicitly as well, which is wrong.

erimatnor replied:
Can we even have a default value for non-timestamp-like columns?

If the dimension type is something like `date` or `timestamp` we know the time unit (e.g., microseconds for timestamps). But if it is an integer type, we cannot know the unit (time resolution). It could be seconds, milliseconds, or whatever. Therefore, must we not force the user to set `chunk_time_interval` in case of integer type column?

mfreed replied:
I somewhat like that idea.  As @erimatnor points out, integers/bigints are not really semantically typed from the perspective of time, so perhaps requiring an explicit (rather than a default) interval is the way to go.

RobAtticus replied:
So the proposal is to set it to NULL by default, and if it's a TIMESTAMP/TZ we set it to `1 month`, otherwise throw an error saying it needs to be supplied?

mfreed replied:
LGTM.

Also, it appears that our docs actually *already* say that :  "Additionally, if the time column is of type SMALLINT or INT, the `chunk_time_interval` **must** be set explicitly."  So I'm glad the code now matches :)

RobAtticus commented:
I've changed the behavior to what we discussed:

* If the time field is TIMESTAMP/TZ, we set the default to 1 month.
* If the time field is SMALLINT, INTEGER, or BIGINT, we throw an exception saying it needs to be explicitly set.

Also added a check for intervals too large for SMALLINT.

cevian commented on sql/ddl_internal.sql:
```diff
@@ -87,6 +87,7 @@ $BODY$
 DECLARE
     partitioning_func        _timescaledb_catalog.dimension.partitioning_func%TYPE = 'get_partition_for_key';
     partitioning_func_schema _timescaledb_catalog.dimension.partitioning_func_schema%TYPE = '_timescaledb_internal';
+    -- this var is never used? (- rrk)
```
Can you remove it then?

erimatnor commented on sql/ddl_api.sql:
```diff
@@ -81,6 +84,31 @@ BEGIN
+    -- Timestamp types can use default value, integral should be an error if NULL
+    IF time_type IN ('TIMESTAMP', 'TIMESTAMPTZ') AND chunk_time_interval IS NULL THEN
+        chunk_time_interval_actual := _timescaledb_internal.interval_to_usec('1 month');
+    ELSIF time_type IN ('SMALLINT', 'INTEGER', 'BIGINT') AND chunk_time_interval IS NULL THEN
```
should we consider supporting integer types like SERIAL, and BIGSERIAL. And, what about decimal types? Or, should such additions be a separate PR?

RobAtticus replied:
Separate PRs, let's let this one just fix what is currently available.

erimatnor commented on sql/ddl_api.sql:
```diff
@@ -81,6 +84,31 @@ BEGIN
+    WHERE attrelid = main_table AND attname = time_column_name;
+
+    -- Timestamp types can use default value, integral should be an error if NULL
+    IF time_type IN ('TIMESTAMP', 'TIMESTAMPTZ') AND chunk_time_interval IS NULL THEN
```
Add DATE? Or, is that for separate PR?

RobAtticus replied:
Separate PR.

erimatnor commented on sql/ddl_api.sql:
```diff
@@ -17,23 +17,26 @@ CREATE OR REPLACE FUNCTION  create_hypertable(
     associated_schema_name  NAME = NULL,
     associated_table_prefix NAME = NULL,
-    chunk_time_interval     BIGINT =  _timescaledb_internal.interval_to_usec('1 month'),
+    chunk_time_interval     BIGINT = NULL,
```
Is it possible to add a version of this function that takes an interval instead of a BIGINT? That would allow people that use TIMESTAMP dimensions to simply specify their `chunk_time_interval` as `interval '1 week'`. This version of the function would throw an error on INTEGER type columns and point to BIGINT-version.

RobAtticus replied:
Since we can overload, I like this idea potentially. I think it should be a separate PR just to keep this streamline.
